File: rpm.cpp
#include <iostream>
#include <cmath>

using namespace std;

//Nesta função é definido a equação que será resolvida
double eqc(double x) {
    double f;
    f = x/1;
    //f = pow(x,2); //x^2
    //f = sqrt(x);//raiz quadrada de x
    //f = cbrt(x);//raiz cúbica de x
    return f;
}

//Nesta função é definido o passo-a-passo de como resolver de forma algoritmica a regra do ponto médio
double integral(double a, double b, int n){
    //recebe o valor de delta x
    double dx = (b - a)/n;
    double vet1[n - 1];
    double vet2[n];
    double sum = 0;
    double result = 0;
    //armazena os valores da primeira divisão de pontos em vet1
    cout << "Primeira divisao" << endl;
    for (int i = 1; i < n; ++i) {
    //este primeiro if existe para definir o primeiro ponto depois de 'a' (sendo 'a' o primeiro ponto no intervalo ('a', 'b'))
        if (i == 1){
            vet1[i] = a + dx;
            cout << vet1[i] << endl;
        } else{
            vet1[i] = vet1[i-1] + dx;
            cout << vet1[i] << endl;
        }
    }
    //armazenas os valores dos pontos médios
    cout << "Segunda divisao" << endl;
    //pm define o valor que vai servir pra definir os pontos médios
    double pm = (vet1[1] - a)/2;
    for (int j = 1; j <= n; ++j) {
    //este primeiro if existe para definir o primeiro ponto médio após 'a' (sendo 'a' o primeiro ponto no intervalo ('a', 'b'))
        if (j == 1){
            vet2[j] = a + pm;
            cout << vet2[j] << endl;
        } else{
            vet2[j] = vet1[j-1] + pm;
            cout << vet2[j] << endl;
        }
        //neste ponto é realiza o somatório de todos os pontos médios na equação(que foi definida na função eqc(x))
        //onde o valor de x vai ser igual ao valor armazenado em vet2[j]
        sum = sum + eqc(vet2[j]);
    }
//-----------------Esta parte é puro enfeite---------------------------------------------------------
    //Exibe todas as divisões em uma "reta"
    for (int k = 1; k < n; ++k) {
        if(k == 1){
            cout << a << "--" << vet2[k] << "--" << vet1[k] << "--";
        }else{
            cout << vet2[k] << "--" << vet1[k] << "--";
        }
    }
    cout << b << endl;
//----------------Acaba aqui o enfeite---------------------------------------------------------------
    //multiplica delta x com o somatorio
    result = dx * sum;
    //cout.precision(10); define a quantidade de casas decimais após a vírgula
    cout.precision(10);
    //exibe o resultado
    cout << "Resultado: " << fixed << result << endl;
}
//Basta declarar a função 'integral'. Dentro do parenteses as variáveis são (a, b, n), onde, 'a' e 'b'
//definem o intervalo e 'n' o número de divisões.
//O main aqui é utilizado apenas para receber a estrela do show, a função integral(a, b, n)
int main() {
    integral(1, 2, 5);
    return 0;
}

File: rpm.py
#Aqui é declarado a função a ser resolvida
def func (x):
    from math import exp, pow
    #f = 1/x
    f = float((pow(x,2))*exp(-x))
    return f
#Aqui é onde acontece tudo, "a" e "b" são os intervalos definidos e "n" é o número de divisões
def integral(a, b, n):
    #Calculo do delta x
    dx = (b-a)/n
    #Declaração da lista que irá receber os valores dos pontos médios
    lista = []
    #"pm1" é o primeiro ponto médio encontrado... ele é tratado agora para não dar dor de cabeça depois
    pm1 = (dx / 2) + a
    #"pm1" é inserido na lista
    lista.append(pm1)
    #"soma" será a variável responsável pelo somatório de todos os pontos médios na f(x)
    #"func(pm1)" indica que o valor recebido por "soma" já é o resultado do calculo do "pm1" na função
    soma = func(pm1)
    #Aqui "i" é incializado em 1 
    for i in range(1, n):
        #Aqui é adicionado na lista os pontos médios
        #Ponto anterior mais o valor de dx
        lista.append(lista[i-1]+dx)
        #Aqui é adicionado na soma os valores dos pontos seguintes aplicados na função definida
        soma +=func(lista[i])
    #"result" recebe a multiplicação de delta x com a soma de todos os pontos médios já aplicados na função
    result = dx*soma
    #Enfeites de impressão
    #print("lista =", lista)
    #print("soma =", soma)
    print("resultado =", result)
integral(0, 1000, 100000)